bin/asl-node-lookup

```diff
@@ -0,0 +1,266 @@
+    if [[ -x /usr/bin/apt ]]; then
+	echo ""
+	echo "To install the \"host\" command, use \"apt install bind9-dnsutils\"."
+    fi
+    exit 1
+fi
+
+# ---------- ---------- ---------- ---------- ----------
+
+usage()
+{
+    echo "Usage: $0 [ --help ] [ --verbose ] [ --ns <name-server> ] <node#>"
+}
+
+do_soa()
+{
+    SOA_HOST=$1
+
+    #
+    # dns SOA record
+    #
+    SOA="$(host -t SOA "${SOA_HOST}" ${USE_NS}		\
+	   | grep -v					\
+		  -e "^Using domain server:"		\
+		  -e "^Name:"				\
+		  -e "^Address:"			\
+		  -e "^Aliases:"			\
+		  -e "^$"				\
+	  )"
+    echo ""
+    echo "SOA ($SOA_HOST)"
+    echo "${SOA}"					\
+	| sed -e "s/^${SOA_HOST} has SOA record //"	\
+	      -e "s/^/  /"
+}
+
+do_ns()
+{
+    NS_HOST=$1
+
+    #
+    # dns NS record
+    #
+    NS="$(host -t NS "${NS_HOST}" ${USE_NS}		\
+	  | grep -v					\
+		 -e "^Using domain server:"		\
+		 -e "^Name:"				\
+		 -e "^Address:"				\
+		 -e "^Aliases:"				\
+		 -e "^$"				\
+	 )"
+    echo ""
+    echo "NS ($NS_HOST)"
+    echo "${NS}"					\
+	| sed -e "s/^/  /"
+}
+
+do_srv()
+{
+    SRV_HOST="_iax._udp.${1}.nodes.allstarlink.org"
+
+    #
+    # node SRV record
+    #
+    SRV="$(host -t SRV "${SRV_HOST}" ${USE_NS}		\
+	   | grep -v					\
+		  -e "^Using domain server:"		\
+		  -e "^Name:"				\
+		  -e "^Address:"			\
+		  -e "^Aliases:"			\
+		  -e "^$"				\
+	  )"
+    echo ""
+    echo "SRV ($SRV_HOST)"
+    re="^${SRV_HOST} has SRV record"
+    if [[ ${SRV} =~ $re ]]; then
+	echo "${SRV}"					\
+	| sed -e "s/^${SRV_HOST} has SRV record //"	\
+	      -e "s/^/  /"
+    else
+	# report no SRV
+	echo "  No SRV record"
+	SRV=""
+    fi
+}
+
+do_a()
+{
+    HOST=$1
+
+    #
+    # node A record
+    #
+    IP4="$(host -t A "${HOST}" ${USE_NS}		\
+	   | grep -v					\
+		  -e "^Using domain server:"		\
+		  -e "^Name:"				\
+		  -e "^Address:"			\
+		  -e "^Aliases:"			\
+		  -e "^$"				\
+	  )"
+    echo ""
+    echo "A ($HOST)"
+    re="^${HOST} has address"
+    if [[ ${IP4} =~ $re ]]; then
+	echo "${IP4}"					\
+	| sed -e "s/^${HOST} has address //"		\
+	      -e "s/^/  /"
+    else
+	# report no address
+	echo "  No address (A) record"
+    fi
+}
+
+do_txt()
+{
+    HOST=$1
+
+    #
+    # node TXT record
+    #
+    TXT="$(host -t TXT ${HOST} ${USE_NS}		\
+	   | grep -v					\
+		  -e "^Using domain server:"		\
+		  -e "^Name:"				\
+		  -e "^Address:"			\
+		  -e "^Aliases:"			\
+		  -e "^$"				\
+	  )"
+    echo ""
+    echo "TXT ($HOST)"
+    re="^${HOST} descriptive text"
+    if [[ ${TXT} =~ $re ]]; then
+	echo "${TXT}"					\
+	| sed -e "s/^${HOST} descriptive text //"	\
+	      -e 's/^"/  /'				\
+	      -e 's/" "/\n    /g'			\
+	      -e 's/"$//'
+    else
+	# report no TXT
+	echo "  No TXT record"
+    fi
+}
+
+do_rpt_lookup()
+{
+    ASL_NODE=$1
+
+    RPT="$(asterisk -x "rpt lookup ${ASL_NODE}"				\
+	   | sed -E							\
+		 -e 's/.*(radio@)/  \1/'				\
+		 -e 's/ *$//'						\
+		 -e 's/([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)(\/)/\1:4569\2/'	\
```

Comment:
Why is this command hardcoding the port number `:4569`? I don't think there is a guarantee that they have the node configured to that port, it's just the default.

Reply:
That edit will ADD the port if it's not already present with the end result being string that includes both the IP address and the port.

Reply:
Yes, but the issue is the `sed` command is not going to respect the actual node configuration. When there is a custom port the information presented would be false. What is the need to add this into the output instead of displaying more precisely what comes back from the lookup?

Reply:
I see the confusion.  In a version from earlier today I was taking the value from `rpt lookup <node>` and extracting the IP and port.  In order to ensure that a "port" was present in the string I was adding ":4569" IF it was missing (the port is optional, defaulting to 4569).  I did that to simplify my extraction.  I then changed the reported output to hand back what was returned.  So, I guess I really don't need to add the missing port.

Reply:
Fixed!